dali/python/nvidia/dali/plugin/jax/iterator.py

```diff
@@ -303,9 +308,17 @@ def create_iterator(*args, **wrapper_kwargs):
                 # Handle batch_size_per_gpu
                 global_batch_size = wrapper_kwargs["batch_size"]
                 batch_size_per_gpu = global_batch_size // len(jax.devices())
+
                 wrapper_kwargs["batch_size"] = batch_size_per_gpu
 
-                devices_to_use = devices if devices is not None else jax.devices()
+                devices_to_use = devices if devices is not None else jax.local_devices()
+                num_shards = len(devices) if devices is not None else jax.device_count()
+
+                if devices is not None:
+                    assert jax.local_device_count() == jax.device_count(), (
+                        "Iterator compatible with pmapped JAX functions does not support "
+                        "running in multiprocess mode. Use `sharding` argument instead."
+                    )
```

Comment:
Assertions != error checking. If you intend this to be a proper error, please use appropriate exception with explicit raise.

Reply:
Done